test/test_transforms_v2_refactored.py

```diff
@@ -3859,6 +3859,776 @@ def test_correctness_perspective_bounding_boxes(self, startpoints, endpoints, fo
+
+    @pytest.mark.parametrize(
+        ("kernel", "input_type"),
+        [
+            (F.adjust_gamma_image, torch.Tensor),
+            (F._adjust_gamma_image_pil, PIL.Image.Image),
+            (F.adjust_gamma_image, tv_tensors.Image),
+            (F.adjust_gamma_video, tv_tensors.Video),
+        ],
+    )
+    def test_functional_signature(self, kernel, input_type):
+        check_functional_kernel_signature_match(F.adjust_gamma, kernel=kernel, input_type=input_type)
+
+    def test_functional_error(self):
+        with pytest.raises(ValueError, match="Gamma should be a non-negative real number"):
+            F.adjust_gamma(make_image(), gamma=-1)
+
+    @pytest.mark.parametrize("gamma", [0.1, 0.5, 1.0])
+    @pytest.mark.parametrize("gain", [0.1, 1.0, 2.0])
+    def test_correctness_image(self, gamma, gain):
+        image = make_image(dtype=torch.uint8, device="cpu")
+
+        actual = F.adjust_gamma(image, gamma=gamma, gain=gain)
+        expected = F.to_image(F.adjust_gamma(F.to_pil_image(image), gamma=gamma, gain=gain))
+
+        assert_equal(actual, expected)
+
+
+class TestAdjustHue:
+    @pytest.mark.parametrize("dtype", [torch.uint8, torch.float32])
+    @pytest.mark.parametrize("device", cpu_and_cuda())
+    def test_kernel_image(self, dtype, device):
+        check_kernel(F.adjust_hue_image, make_image(dtype=dtype, device=device), hue_factor=0.25)
+
+    def test_kernel_video(self):
+        check_kernel(F.adjust_hue_video, make_video(), hue_factor=0.25)
+
+    @pytest.mark.parametrize("make_input", [make_image_tensor, make_image, make_image_pil, make_video])
+    def test_functional(self, make_input):
+        check_functional(F.adjust_hue, make_input(), hue_factor=0.25)
+
+    @pytest.mark.parametrize(
+        ("kernel", "input_type"),
+        [
+            (F.adjust_hue_image, torch.Tensor),
+            (F._adjust_hue_image_pil, PIL.Image.Image),
+            (F.adjust_hue_image, tv_tensors.Image),
+            (F.adjust_hue_video, tv_tensors.Video),
+        ],
+    )
+    def test_functional_signature(self, kernel, input_type):
+        check_functional_kernel_signature_match(F.adjust_hue, kernel=kernel, input_type=input_type)
+
+    def test_functional_error(self):
+        with pytest.raises(TypeError, match="permitted channel values are 1 or 3"):
+            F.adjust_hue(make_image(color_space="RGBA"), hue_factor=0.25)
+
+        for hue_factor in [-1, 1]:
+            with pytest.raises(ValueError, match=re.escape("is not in [-0.5, 0.5]")):
+                F.adjust_hue(make_image(), hue_factor=hue_factor)
+
+    @pytest.mark.parametrize("hue_factor", [-0.5, -0.3, 0.0, 0.2, 0.5])
+    def test_correctness_image(self, hue_factor):
+        image = make_image(dtype=torch.uint8, device="cpu")
+
+        actual = F.adjust_hue(image, hue_factor=hue_factor)
+        expected = F.to_image(F.adjust_hue(F.to_pil_image(image), hue_factor=hue_factor))
+
+        mae = (actual.float() - expected.float()).abs().mean()
```

Comment:
Not sure if it's worth it's relevant in this case, but it may be worth checking if we can write similar checks as in https://github.com/pytorch/vision/pull/7943#discussion_r1319849466

Reply:
I tried, but I couldn't find any configuration for the other two thresholds that would provide any additional safety. For example `assert (abs_diff > 2).float().mean() < 20e-2` or `assert (abs_diff > 5).float().mean() < 5e-2` do not hold.